plugins/aws/resoto_plugin_aws/resource/sqs.py

```diff
@@ -0,0 +1,89 @@
+        "arn": S("QueueArn"),
+        "sqs_queue_url": S("QueueUrl"),
+        "sqs_approximate_number_of_messages": S("ApproximateNumberOfMessages"),
+        "sqs_approximate_number_of_messages_not_visible": S("ApproximateNumberOfMessagesNotVisible"),
+        "sqs_approximate_number_of_messages_delayed": S("ApproximateNumberOfMessagesDelayed"),
+        "sqs_policy": S("Policy"),
+        "sqs_redrive_policy": S("RedrivePolicy"),
+        "sqs_fifo_queue": S("FifoQueue"),
+        "sqs_content_based_deduplication": S("ContentBasedDeduplication"),
+        "sqs_kms_master_key_id": S("KmsMasterKeyId"),
+        "sqs_kms_data_key_reuse_period_seconds": S("KmsDataKeyReusePeriodSeconds"),
+        "sqs_deduplication_scope": S("DeduplicationScope"),
+        "sqs_fifo_throughput_limit": S("FifoThroughputLimit"),
+        "sqs_redrive_allow_policy": S("RedriveAllowPolicy"),
+        "sqs_visibility_timeout": S("VisibilityTimeout"),
+        "sqs_maximum_message_size": S("MaximumMessageSize"),
+        "sqs_message_retention_period": S("MessageRetentionPeriod"),
+        "sqs_delay_seconds": S("DelaySeconds"),
+        "sqs_receive_message_wait_time_seconds": S("ReceiveMessageWaitTimeSeconds"),
+        "sqs_managed_sse_enabled": S("SqsManagedSseEnabled"),
+    }
+    sqs_queue_url: str = field(default=None)
+    sqs_approximate_number_of_messages: Optional[int] = field(default=None)
+    sqs_approximate_number_of_messages_not_visible: Optional[int] = field(default=None)
+    sqs_approximate_number_of_messages_delayed: Optional[int] = field(default=None)
+    sqs_policy: Optional[str] = field(default=None)
+    sqs_redrive_policy: Optional[str] = field(default=None)
+    sqs_fifo_queue: Optional[bool] = field(default=None)
+    sqs_content_based_deduplication: Optional[bool] = field(default=None)
+    sqs_kms_data_key_reuse_period_seconds: Optional[int] = field(default=None)
+    sqs_deduplication_scope: Optional[str] = field(default=None)
+    sqs_fifo_throughput_limit: Optional[str] = field(default=None)
+    sqs_redrive_allow_policy: Optional[str] = field(default=None)
+    sqs_visibility_timeout: Optional[int] = field(default=None)
+    sqs_maximum_message_size: Optional[int] = field(default=None)
+    sqs_message_retention_period: Optional[int] = field(default=None)
+    sqs_delay_seconds: Optional[int] = field(default=None)
+    sqs_receive_message_wait_time_seconds: Optional[int] = field(default=None)
+    sqs_managed_sse_enabled: Optional[bool] = field(default=None)
+
+    @classmethod
+    def collect(cls: Type[AwsResource], json: List[Json], builder: GraphBuilder) -> None:
+        def add_tags(queue: AwsSqsQueue) -> None:
+            tags = builder.client.list("sqs", "list-queue-tags", result_name="Tags", QueueUrl=[queue.sqs_queue_url])
+            if tags:
+                queue.tags = cast(Dict[str, Optional[str]], tags)
+
+        for queue_url in json:
+            queue_attributes = builder.client.call(
```

Comment:
Wdyt about not doing this call here, but rather using an approach similar to add_tags? this way we won't block the current thread for the http call.

Reply:
I don't get an approach like that to work . With only the `QueueUrl` from the first `list-queues` api call it's not possible to create an instance with `instance = cls.from_api(js)` 🤔

Reply:
I was thinking about something like:
```
for queue_url in json:
    builder.submit_work(add_instance, queue_url)
    ...
 ```
 
 and move the content of the existing loop into the `add_instance` method